### PR TITLE
test(git-browse): add unit tests

### LINKED ISSUE #1125: Test(git-browse): unit tests
## File
[git-browse](https://github.com/tj/git-extras/blob/main/bin/git-browse)

`git browse` is that the command browses the files of the repository which comes from gitlab, github, bitbucket or other sites.

## Testcases
1. If the repository comes from gitlab, open a url like `https://gitlab.com/<user_or_group>/<repo>/-/blob/<commit_or_branch>/<filename>#L<line1>-<line2>`
2. If the repository comes from github, open a url like `https://github.com/<user_or_org>/<repo>/blob/<commit_or_branch>/<filename>#L<line1>-L<line2>`
3. If the repository comes from bitbucket, open a url like `https://bitbucket.org/<user_or_org>/<repo>/src/<commit_or_branch>/<filename>#lines-<line1>:<line2>`
4. If the repository comes from other sites, open a url to the `origin` and cannot locate to something about line of one file
5. If the OS is mac, open via command `open`
6. If the OS is Git-Bash on windows, open via command `start`
7. If the OS is WSL on windows, open via commnd `start`
8. If the OS is linux, open via command `xdg-open`
9. If the OS is unknown, open via command `xdg-open`

### PR DESCRIPTION
**Pull request by vanpipy**
close #1125

### CODE REVIEW

**Comment by hyperupcall** [comment]
LGTM

**Review by hyperupcall — APPROVED** [review]

File: tests/test_git_browse.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
+import os, subprocess
+from testpath import MockCommand, modified_env
+
+github_origin = "https://github.com/vanpipy/git-extras"
```
Comment (spacewander) [inline tests/test_git_browse.py:4]: Is it possible to use `tj/git-extras` in the github test case?

Comment (vanpipy) [reply tests/test_git_browse.py]: Update

**Comment by vanpipy** [comment]
Please squash and merge this PR cause the I did not squash the commits manually to avoid the force pushing.

**Review by spacewander — APPROVED** [review]

**Comment by spacewander** [comment]
@vanpipy 
Thanks!

**Pull request merged**